dev/ansible/01_generate_secrets_vault.yml

```diff
@@ -0,0 +1,29 @@
+- name: Generate Secrets with Vault
+  shell: |
+      vault kv put secret/postgres \
+      username=$(vault write -field='value' gen/password length=12 symbols=0) \
+      password=$(vault write -field='value' gen/password length=12 symbols=0)
+  run_once: true
+  environment:
+      VAULT_ADDR: "http://127.0.0.1:8200"
+
+- name: Register username key
+  shell:
+      cmd: vault kv get -field='username' secret/postgres
+  register: username
+
+- name: Register password key
+  shell:
+      cmd: vault kv get -field='password' secret/postgres
+  register: password
+
+- name: Delete /etc/default/terraform.tfvars if it exists
+  file:
+      path: /etc/default/terraform.tfvars
+      state: absent
+
+- name: Add username and password to the /etc/default/terraform.tfvars file
+  shell: |
+      echo username='"{{ username.stdout }}"' >> /etc/default/terraform.tfvars
+      echo password='"{{ password.stdout }}"' >> /etc/default/terraform.tfvars
+  run_once: true
```

Comment:
If we do this from terraform we do not need to create any extra files.

I realise everything I comment ATM was also present in Neha's PR, sorry about that. I glossed over it too quickly it seems!

Reply:
How do we do it with terraform?

Reply:
You can use a `vault_generic_secret`. Hold on, I'll edit this comment with an example (there is one in the code)